src/mars.d

```diff
@@ -1575,6 +1578,28 @@ Language changes listed by -transition=id:
             gendocfile(m);
         }
     }
+    if(global.params.vcg_ast)
+    {
+        import ddmd.hdrgen;
+        foreach(mod;modules)
+        {
+            auto buf = new OutBuffer;
+            buf.doindent = 1;
+            scope HdrGenState hgs;
+            hgs.fullDump = 1;
+            scope PrettyPrintVisitor ppv = new PrettyPrintVisitor(buf, &hgs);
+            mod.accept(ppv);
+
+            auto modFilename = mod.srcfile.toChars();
+            auto modFilenameLength = strlen(modFilename);
+            auto cgFilename = cast(char*)allocmemory(modFilenameLength + 4);
+            strcpy(cgFilename, modFilename);
+            cgFilename[modFilenameLength .. modFilenameLength + 4] = ".cg\0";
```

Comment:
I'm not sure what that code is doing to the filename, but in root.filename there are `forceExt`, `defaultExt`, etc.